cs-base-station-interface/src/main/java/basestation/bot/BotManager.java

```diff
@@ -89,4 +90,23 @@ public int generateBotNumber() {
     public Set<String> getAllDiscoveredBots() {
         return udpConnection.getAddressSet();
     }
+
+    /**
+     * Returns the IP associated with Bot IP Mapping
+     *
+     * @return String
+     */
+    public String getBotIP (String id) {
+        return botIPMap.get(id);
+    }
+
+    /**
+     * Add mapping from this bot's id to its ip
+     * @param id name specified on request from bot
+     * @param IP IP given by the bot
+     * @return true if operation was successful
+     */
+    public boolean setBotIPMap (String id, String IP) {
+        return botIPMap.put(id, IP) != null;
```

Comment:
can't you get this IP from the existing connection with the bot? It looks like this is more like an arbitrary key/value storer so it's a bit limiting to say this is just for IPs